apps/next-evans/lib/dates.ts

```diff
@@ -29,9 +29,19 @@ export const months = {
   ],
 };
 
-export function newestFirst(
-  a: Date | string | number,
-  b: Date | string | number,
-): number {
-  return new Date(b).getTime() - new Date(a).getTime();
+export function newestFirst(a: number, b: number): number;
+export function newestFirst(a: ISODateString, b: ISODateString): number;
+export function newestFirst<T extends { createdAt: ISODateString }>(a: T, b: T): number;
```

Comment:
a more flexible signature (and hence, a changed implementation below) allows us to use this in most places as just `.sort(newestFirst)` which appeals to the functional purist in me.